python/ray/tests/conftest_docker.py

```diff
@@ -141,3 +144,76 @@ def print_logs(self):
 @pytest.fixture
 def docker_cluster(head_node, worker_node):
     yield (head_node, worker_node)
+
+
+def run_in_container(cmd: List[str], container_id: str):
+    docker_cmd = ["docker", "exec", container_id] + cmd
+    print(f"Executing command: {docker_cmd}", time.time())
+    resp = subprocess.check_output(docker_cmd, stderr=subprocess.STDOUT, timeout=1000)
+    output = resp.decode("utf-8").strip()
+    print(f"Output: {output}")
+    return output
+
+
+IMAGE_NAME = "rayproject/ray:runtime_env_container"
+NESTED_IMAGE_NAME = "rayproject/ray:runtime_env_container_nested"
+
+
+@pytest.fixture
+def podman_docker_cluster():
+    start_container_command = [
+        "docker",
+        "run",
+        "-d",
+        "--privileged",
+        "-v",
+        "/var/run/docker.sock:/var/run/docker.sock",
+        "-v",
+        "/var/lib/containers:/var/lib/containers",
+        IMAGE_NAME,
+        "tail",
+        "-f",
+        "/dev/null",
+    ]
+    container_id = subprocess.check_output(start_container_command).decode("utf-8")
+    container_id = container_id.strip()
+
+    # Get group id that owns the docker socket file. Add user `ray` to
+    # group to get necessary permissions for pulling an image from
+    # docker's local storage into podman
+    ls_output = run_in_container(["ls", "-l", "/var/run/docker.sock"], container_id)
+    regex = ".{10} +\d+ +root +(\d+) \d+ [A-Za-z]+ +\d+ +.+ +\/var\/run\/docker\.sock"
+    docker_group_id = re.search(regex, ls_output).group(1)
+    run_in_container(["id"], container_id)  # For debugging
+    run_in_container(
+        ["sudo", "groupadd", "-g", docker_group_id, "docker"], container_id
+    )
+    run_in_container(["sudo", "usermod", "-aG", "daemon", "ray"], container_id)
+    run_in_container(["sudo", "usermod", "-aG", "docker", "ray"], container_id)
+
+    # Pull image from docker's local storage into podman
+    run_in_container(["podman", "pull", f"docker-daemon:{IMAGE_NAME}"], container_id)
+
+    # Add custom file to new image tagged `runtime_env_container_nested`,
+    # which can be read by Ray actors / Serve deployments to verify the
+    # container runtime env plugin
+    run_in_container(
+        ["bash", "-c", "echo helloworldalice >> /tmp/file.txt"], container_id
+    )
+    run_in_container(
+        ["podman", "create", "--name", "tmp_container", IMAGE_NAME], container_id
+    )
+    run_in_container(
+        ["podman", "cp", "/tmp/file.txt", "tmp_container:/home/ray/file.txt"],
+        container_id,
+    )
+    run_in_container(
+        ["podman", "commit", "tmp_container", NESTED_IMAGE_NAME], container_id
+    )
```

Comment:
nit: use a for loop over a list of list, so that you do not need to repeat the `container_id`?